index.js

```diff
@@ -13,10 +13,13 @@ app.get("/secret/:id", (req, res) => {
   const id = req.params.id;
 
   if (secrets[id]) {
-    res.send(secrets[id]);
     //check if we should delete
-    if (secrets[id].timeout === 0) {
+    if (secrets[id].expiry < Date.now()) {
```

Comment:
We only delete expired secrets when they are requested. This means expired secrets can pile up and ideally are purged in a regular, `cron`-style cleaning job